### ISSUE TITLE
selectizeInput(): certain suggestions are truncated when typing input

### ISSUE THREAD

**Issue by flolauck** [rstudio/shiny#3552]
### System details

Browser Version: Chrome 96.0.4664.45

Output of `sessionInfo()`:

```
R version 4.0.4 (2021-02-15)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: Ubuntu 20.04.2 LTS

Matrix products: default
BLAS/LAPACK: /software/R-v4.0.4/lib/libopenblasp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] shiny_1.6.0.9000

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6          withr_2.4.1         digest_0.6.27       later_1.1.0.1       mime_0.10           R6_2.5.0           
 [7] jsonlite_1.7.2      lifecycle_1.0.0     xtable_1.8-4        magrittr_2.0.1      cachem_1.0.4        rlang_0.4.10       
[13] promises_1.2.0.1    jquerylib_0.1.3     bslib_0.2.4         ellipsis_0.3.1      BiocParallel_1.24.1 tools_4.0.4        
[19] fastmap_1.1.0       httpuv_1.5.5        yaml_2.2.1          parallel_4.0.4      compiler_4.0.4      htmltools_0.5.1.1  
[25] sass_0.3.1 
```

### Example application *or* steps to reproduce the problem

 

```R
library(shiny)

filters <- c(
"~ Issue",
"~ Sprint",
"~ Project",
"~ Project:Sprint",
"~ Project + Sprint + Project:Sprint",
"~ Project + Issue + Project:Issue",
"~ Sprint + Issue + Sprint:Issue",
"~ Issue + Sprint + Issue:Sprint"
)

ui <- fluidPage(
  selectizeInput("filters", label = "Filters", choices = filters),
)

server <- function(input, output, session) {
}

shinyApp(ui, server)
```

### Describe the problem in detail

- Select the Filters field
- Delete the default entry
- Start typing "Issue + Sprint"
  ![shiny-1](https://user-images.githubusercontent.com/2071858/142838425-62a2beb6-e8ed-4ac7-ad1d-b0cb475f72d5.png)
- After typing the letter "p" of "Sprint" the word "Issue" is truncated to "I" in the list of suggestions.
   ![shiny-2](https://user-images.githubusercontent.com/2071858/142838430-ee475c37-b42f-4078-b864-2d0f41390ac8.png)

The problem does not occur for other search strings, e.g. "Project + Sprint".

### Conjecture

The problem seems to be in the JS code, not R. This is the resulting div element: The data-value attribute seems fine, but the content of the div is messed up.
```HTML
<div class="option" data-selectable="" data-value="~ Sprint + Issue + Sprint:Issue" role="option" id="mc9c28031i">~ <span class="highlight">Sp</span>rint <span class="highlight">+</span> I <span class="highlight">+</span> <span class="highlight">Sp</span>rint:I</div>
```

**Comment by gsmolinski** [comment]
I would like to add (or maybe this is obvious?) that this problem occurs also for `selectInput` and is present in `shiny 1.7.1`. This is example I have seen (the same as described by @flolauck, delete the default option and start typing "Środa Śl" - when you press "l" (small "L" letter), you will see truncated name:

```
library(shiny)

ui <- fluidPage(
  selectInput("test", label = NULL, choices = "Środa Śląska")
)

server <- function(input, output, session) {
  
}

shinyApp(ui, server)
```

```
sessionInfo()
#> R version 4.1.1 (2021-08-10)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 10 x64 (build 19042)
#> 
#> Matrix products: default
#> 
#> locale:
#> [1] LC_COLLATE=Polish_Poland.1250  LC_CTYPE=Polish_Poland.1250   
#> [3] LC_MONETARY=Polish_Poland.1250 LC_NUMERIC=C                  
#> [5] LC_TIME=Polish_Poland.1250    
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] shiny_1.7.1
#> 
#> loaded via a namespace (and not attached):
#>  [1] Rcpp_1.0.7        pillar_1.6.3      compiler_4.1.1    later_1.3.0      
#>  [5] highr_0.9         R.methodsS3_1.8.1 R.utils_2.11.0    tools_4.1.1      
#>  [9] digest_0.6.28     evaluate_0.14     lifecycle_1.0.1   tibble_3.1.5     
#> [13] R.cache_0.15.0    pkgconfig_2.0.3   rlang_0.4.11      reprex_2.0.1     
#> [17] yaml_2.2.1        xfun_0.26         fastmap_1.1.0     withr_2.4.2      
#> [21] styler_1.6.2      stringr_1.4.0     knitr_1.36        fs_1.5.0         
#> [25] vctrs_0.3.8       glue_1.4.2        R6_2.5.1          fansi_0.5.0      
#> [29] rmarkdown_2.11    purrr_0.3.4       magrittr_2.0.1    backports_1.2.1  
#> [33] promises_1.2.0.1  ellipsis_0.3.2    htmltools_0.5.2   xtable_1.8-4     
#> [37] mime_0.12         httpuv_1.6.3      utf8_1.2.2        stringi_1.7.5    
#> [41] crayon_1.4.1      R.oo_1.24.0
```

**Comment by gsmolinski** [comment]
I confirm that for at least this usecase:

```
library(shiny)

ui <- fluidPage(
  selectInput("test", label = NULL, choices = "Środa Śląska")
)

server <- function(input, output, session) {
  
}

shinyApp(ui, server)
```
**Problem is solved :)**

Thank you!

**Comment by flolauck** [comment]
For what it's worth, it is fixed for me as well when I use 1.8.0.

Thank you! 😊
